### PR TITLE
fix(#979): fix namespace when installed by helm

### LINKED ISSUE #979: Specify namespace when install Cyclone by Helm
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

```sh
helm install ./helm/cyclone
Error: release plinking-mongoose failed: namespaces "cyclone-system" not found
```

**What you expected to happen**:

It is recommended to use `helm install` with the `--namespace=<namespace_name>` option 
 then Helm should create a namespace for you automatically. You can reference that namespace in your chart with {{ .Release.Namespace }}. 

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #979

Reference to #

**Special notes for your reviewer**:

/cc @cd1989  @zhujian7 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/workflow/controller/Dockerfile [hunk 1]
```diff
@@ -6,4 +6,4 @@ WORKDIR /workspace
 
 COPY ./bin/workflow/controller /workspace/controller
 
-CMD ["./controller"]
\ No newline at end of file
+ENTRYPOINT ["./controller"]
```
Comment (cd1989) [inline build/workflow/controller/Dockerfile]: Why use ENTRYPOINT ?

Comment (supereagle) [reply build/workflow/controller/Dockerfile]: Need to pass `--namespace` flag. Refer to  https://github.com/caicloud/cyclone/pull/981/files#diff-f31db4d87194712918aaca8ebba3a978R82

File: helm/cyclone/values.yaml [hunk 2]
```diff
@@ -2,15 +2,14 @@
 # while 'libraryProject' specifies where to pull common images like 'alpine', 'busybox'.
 # These two projects can be definitely set to the same value.
 imageRegistry:
+#  registry: docker.io
+#  project: k8sdevops
```
Comment (cd1989) [inline helm/cyclone/values.yaml]: ?

File: helm/cyclone/templates/server.yaml [hunk 3]
```diff
@@ -77,6 +77,9 @@ spec:
       - name: server
         image: {{ .Values.imageRegistry.registry }}/{{ .Values.imageRegistry.project }}/cyclone-server:{{ .Chart.AppVersion }}
         imagePullPolicy: IfNotPresent
+        args:
+        - "--namespace"
+        - {{ .Release.Namespace }}
```
Comment (cd1989) [inline helm/cyclone/templates/server.yaml]: What's the purpose of `namespace` to server ?

Comment (zhujian7) [reply helm/cyclone/templates/server.yaml]: The server will get cyclone server configmap from this namespace.

File: build/workflow/controller/Dockerfile (hunk 1 above)
Comment (cd1989) [reply build/workflow/controller/Dockerfile]: This can be removed

File: helm/cyclone/values.yaml (hunk 2 above)
Comment (supereagle) [reply helm/cyclone/values.yaml]: Mistake, will revert

File: build/workflow/controller/Dockerfile (hunk 1 above)
Comment (supereagle) [reply build/workflow/controller/Dockerfile]: Fixed

**Comment by supereagle** [comment]
Have specified the version requirement for Helm to fix CRD installation.

> CRDs are not installed, need to check the version of Helm
> 
> /hold

Have fixed comments, PTAL @cd1989 @zhujian7 
/hold cancel

File: cmd/server/main.go [hunk 4]
```diff
@@ -98,7 +95,7 @@ func initialize(opts *Options) {
 	}
 
 	if config.Config.CreateBuiltinTemplates {
-		templates.InitStageTemplates(client, config.GetSystemNamespace(), "")
+		templates.InitStageTemplates(client, config.Config.SystemNamespace, "")
```
Comment (cd1989) [inline cmd/server/main.go]: Why not `config.GetSystemNamespace()` ?

**Pull request merged**